src/test/java/com/google/devtools/build/lib/remote/SimpleBlobStoreFactoryTest.java

```diff
@@ -0,0 +1,190 @@
+
+    assertThat(blobStore).isNotNull();
+    assertThat(blobStore).isInstanceOf(CombinedDiskHttpBlobStore.class);
+    verify(relativeDirectory, times(1)).createDirectoryAndParents();
+  }
+
+  @Test
+  public void createCombinedCache_whenWorkingDirectoryNull_throwsNPE() throws IOException {
+    // interesting case: workingDirectory = null -> NPE.
+    RemoteOptions options = Options.getDefaults(RemoteOptions.class);
+    options.remoteHttpCache = "http://doesnotesist.com";
+    options.diskCache = PathFragment.create("/etc/something/cache/here");
+
+    expectedEx.expect(NullPointerException.class);
+
+    SimpleBlobStoreFactory.create(options, /* creds */ null, /* workingDirectory */ null);
+  }
+
+  @Test
+  public void createHttpCacheWithProxy() throws IOException {
+    RemoteOptions options = Options.getDefaults(RemoteOptions.class);
+    options.remoteHttpCache = "http://doesnotesist.com";
+    options.remoteCacheProxy = "unix://some-proxy";
+    options.diskCache = null;
+
+    SimpleBlobStore blobStore = SimpleBlobStoreFactory.create(options, /* creds */ null, workingDirectory);
+
+    assertThat(blobStore).isNotNull();
+    assertThat(blobStore).isInstanceOf(HttpBlobStore.class);
+  }
+
+  @Test
+  public void createHttpCacheWithProxy_notSupportedProtocol_throwsException() throws IOException {
+    RemoteOptions options = Options.getDefaults(RemoteOptions.class);
+    options.remoteHttpCache = "http://doesnotesist.com";
+    options.remoteCacheProxy = "bad-proxy";
+    options.diskCache = null;
+
+    expectedEx.expect(Exception.class);
+    expectedEx.expectMessage("Remote cache proxy unsupported: bad-proxy");
+
+    SimpleBlobStoreFactory.create(options, /* creds */ null, workingDirectory);
+  }
+
+  @Test
+  public void createHttpCacheWithoutProxy() throws IOException {
+    RemoteOptions options = Options.getDefaults(RemoteOptions.class);
+    options.remoteHttpCache = "http://doesnotesist.com";
+    options.diskCache = null;
+
+    SimpleBlobStore blobStore = SimpleBlobStoreFactory.create(options, /* creds */ null, workingDirectory);
+
+    assertThat(blobStore).isNotNull();
+    assertThat(blobStore).isInstanceOf(HttpBlobStore.class);
+  }
+
+  @Test
+  public void createDiskCache() throws IOException {
+    RemoteOptions options = Options.getDefaults(RemoteOptions.class);
+    options.diskCache = PathFragment.create("/etc/something/cache/here");
+
+    when(workingDirectory.getRelative(options.diskCache)).thenReturn(relativeDirectory);
+    when(relativeDirectory.exists()).thenReturn(true);
+
+    SimpleBlobStore blobStore = SimpleBlobStoreFactory.create(options, /* creds */ null, workingDirectory);
+
+    assertThat(blobStore).isNotNull();
+    assertThat(blobStore).isInstanceOf(OnDiskBlobStore.class);
+  }
+
+  @Test
+  public void isRemoteCacheOptions_true_whenHttpCache() {
+    RemoteOptions options = new RemoteOptions();
```

Comment:
Use Options.getDefaults. That way the fields will be initialized with their default values.

Reply:
done